lib/curve.rb

```diff
@@ -0,0 +1,99 @@
+module EllipticCurve
+
+    module Curve
+
+        class CurveFp
+
+            attr_accessor :a, :b, :p, :n, :gx, :gy, :name, :oid, :nistName
+
+            def initialize(a, b, p, n, gx, gy, name, oid, nistName=nil)
+                @a = a
+                @b = b
+                @p = p
+                @n = n
+                @gx = gx
+                @gy = gy
+                @name = name
+                @oid = oid
+                @nistName = nistName
+            end
+            
+            def contains(p)
+                # Verify if the point `p` is on the curve
+    
+                # :param p: point p = point(x, y)
+                # :return: boolean
+                if not 0 <= p.x && p.x <= @p - 1
+                    return false
+                end
+                if not 0 <= p.y && p.y <= @p - 1
+                    return false
```

Comment:
Nesse caso, aqui a precedência dos operadores garante o mesmo comportamento que no python? Pq o "not" tem que ser aplicado no resultado do "and"
Correto: not (a && b)
Incorreto: (not a) && b

Reply:
para garantir, adicionei os ()